### ISSUE TITLE
Custom SSL certificate

### ISSUE THREAD

**Issue by pSnehanshu** [githubsaturn/captainduckduck#265]
Is it possible to use custom SSL certificate bought from DigiCert, GoDaddy etc.? Let's encrypt may not be a good option for businesses.

**Comment by githubsaturn** [comment]
Yes. You can copy your custom SSL certs in a directory that's being shared inside the nginx container. It requires modification of your nginx config though. See this:
https://captainduckduck.com/docs/nginx-customization.html#custom-files-and-directories

**Comment by Alwinator** [comment]
@githubsaturn Which config should I change to use my custom certificates for the web interface?

**Comment by githubsaturn** [comment]
regular nginx configs: ssl_certificate ssl_certificate_key

**Comment by Alwinator** [comment]
Do you mean `/captain/generated/nginx/nginx.conf` with regular nginx config?

**Comment by githubsaturn** [comment]
No. `generated` directory gets reset ever time your machine reboots. 

Use `/nginx-shared` directory as explained here: https://caprover.com/docs/nginx-customization.html#custom-files-and-directories

**Comment by Alwinator** [comment]
I already have my custom certificates in /nginx-shared, but what to do now? Where is the nginx config in which I should set the certificate to /nginx-shared/my-certificate.pem?

**Comment by githubsaturn** [comment]
Go to your app > and click on Edit Default Nginx Configuration button.

Find this block:
```
    <%
    if (!s.forceSsl) {
    %>
        listen       80;
    <%
    }
    if (s.hasSsl) {
    %>
        listen              443 ssl http2;
        ssl_certificate     <%-s.crtPath%>;
        ssl_certificate_key <%-s.keyPath%>;
    <%
    }
    %>
```
Then, change it to this:
```
    <%
    if (!s.forceSsl) {
    %>
        listen       80;
    <%
    }
    if (true) {                                                                           ### <<< CHANGED
    %>
        listen              443 ssl http2;
        ssl_certificate     /captain/data/nginx-shared/custom-ssl_certificate.crt;        ### <<< CHANGED
        ssl_certificate_key /captain/data/nginx-shared/custom-ssl_certificate_key.key;    ### <<< CHANGED
    <%
    }
    %>
```

**Comment by Alwinator** [comment]
Thanks! That works great for the apps! Now I also want to set the custom certificate for the caprover interface. How can I do that?

**Comment by githubsaturn** [comment]
Same thing, except go to Settings > Edit Default Nginx Config, find this block and change it
```
    <%
        if (captain.hasRootSsl) {
    %>
        listen              443 ssl;
        ssl_certificate     <%-captain.crtPath%>;
        ssl_certificate_key <%-captain.keyPath%>;

    <%
        }
    %>
```

**Comment by Alwinator** [comment]
Thank you very much!

**Comment by githubsaturn** [comment]
See: https://github.com/caprover/caprover/pull/1377

**Comment by BenjaminEHowe** [comment]
@githubsaturn the best practice docs say that the ability to override app default nginx config will be available from v1.11. Is there an estimate for when v1.11 will be released? It looks like the last release was just under a couple of years ago, so I'm hoping soon-ish?

**Comment by baldomerocho** [comment]
> Lo mismo, excepto que vaya a Configuración > Editar configuración predeterminada de Nginx, busque este bloque y cámbielo
> 
> ```
>     <%
>         if (captain.hasRootSsl) {
>     %>
>         listen              443 ssl;
>         ssl_certificate     <%-captain.crtPath%>;
>         ssl_certificate_key <%-captain.keyPath%>;
> 
>     <%
>         }
>     %>
> ```

Works! 🙌

**Comment by xnardo** [comment]
> Go to your app > and click on Edit Default Nginx Configuration button.
> 
> Find this block:
> 
> ```
>     <%
>     if (!s.forceSsl) {
>     %>
>         listen       80;
>     <%
>     }
>     if (s.hasSsl) {
>     %>
>         listen              443 ssl http2;
>         ssl_certificate     <%-s.crtPath%>;
>         ssl_certificate_key <%-s.keyPath%>;
>     <%
>     }
>     %>
> ```
> 
> Then, change it to this:
> 
> ```
>     <%
>     if (!s.forceSsl) {
>     %>
>         listen       80;
>     <%
>     }
>     if (true) {                                                                           ### <<< CHANGED
>     %>
>         listen              443 ssl http2;
>         ssl_certificate     /nginx-shared/custom-ssl_certificate.crt;        ### <<< CHANGED
>         ssl_certificate_key /nginx-shared/custom-ssl_certificate_key.key;    ### <<< CHANGED
>     <%
>     }
>     %>
> ```

I configured this exactly as it is here, with my certificates, but caprover won't let me deploy. He tells me "something bad happened"
